tree/tree/src/TChain.cxx

```diff
@@ -1273,6 +1273,22 @@ Int_t TChain::LoadBaskets(Long64_t /*maxmemory*/)
    return 0;
 }
 
+namespace ROOT::Internal::TChain {
+class TTreeIMTSetterRAII {
+private:
+   TTree *&fTree;
```

Comment:
Why not just?
```
TTree *fTree;
```

Reply:
because the value of the fTree variable can change in the body of the method, e.g. here https://github.com/root-project/root/pull/14691/files#diff-3b55a8f62d4fb5deab481fa4caa779a4180e66bc608396c37365d22e567c8501R1535 .